src/rate_limiter.cc

```diff
@@ -105,29 +109,13 @@ RateLimiter::UnregisterModelInstance(TritonModelInstance* triton_model_instance)
   auto i_it = model_instances.find(triton_model_instance);
   if (i_it != model_instances.end()) {
     if (!ignore_resources_and_priority_) {
-      resource_manager_->RemoveModelInstance(i_it->second.get());
+      RETURN_IF_ERROR(
+          resource_manager_->RemoveModelInstance(i_it->second.get()));
     }
     model_context.RemoveInstance(i_it->second.get());
     model_instances.erase(i_it);
   }
 
-  if (!ignore_resources_and_priority_) {
-    // The update can be skipped if there is only model instance count update,
-    // which new results will not be generated in such case. [FIXME: DLIS-4821]
-    RETURN_IF_ERROR(resource_manager_->UpdateResourceLimits());
-  }
-
-  {
-    std::lock_guard<std::mutex> lk(payload_queues_mu_);
-    auto p_it = payload_queues_.find(model);
-    if (p_it != payload_queues_.end()) {
-      auto s_it = p_it->second->specific_queues_.find(triton_model_instance);
-      if (s_it != p_it->second->specific_queues_.end()) {
-        p_it->second->specific_queues_.erase(s_it);
```

Comment:
I wonder why we are no longer removing `specific_queues`  upon unregistering the model?

Reply:
Thanks for catching. I was thinking it is removing payloads from the specific queue, but it is removing the queue for holding payloads. It is added back: [Revert specific queue erase removal](https://github.com/triton-inference-server/core/pull/208/commits/8fa3e9a3ebff49cfa206baa4712eb335e2734cd0)